### PR TITLE
 #1654 Visual Studio Bug: Output Type of Class Library cannot be started directly

### LINKED ISSUE #1654: Visual Studio Bug: Output Type of Class Library cannot be started directly
photo6183968220466227193

The above bug was encountered while working on PPTLabs.
This is due to errors in Visual Studio's project configuration.
Delete the .vs folder for Visual Studio to recreate the configuration files, fixing the error.

Would be good to document this somewhere in our documentation page in the event another developer faces this problem.

### PR DESCRIPTION
**Pull request by YuPeiHenry**
Fixes #1654

Ready for review.

Added step `Delete the .vs folder in ./PowerPointLabs/PowerPointLabs for Visual Studio to recreate the configuration files.` to ProjectSetUp.md.

Suggestion (for consideration as an alternative):
I have tried a workaround to deleting the .vs. Instead, you can
1. `Right click` *Solution 'PowerPointLabs' (3 projects)* in the solution explorer.
2. Click `Properties`.
3. Navigate to `Common Properties` > `Startup Project`.
4. Select `Single startup project` and choose `PowerPointLabs` from the drop down list.

### CODE REVIEW

**Comment by alexfjw** [comment]
Perhaps there is a better place to put this info? 
If I'm not wrong the bug can appear now and then during development, rather than the first time you setup the IDE.

**Comment by YuPeiHenry** [comment]
Thanks for your suggestion, I have included in Technical Troubleshooting Guide as well, under `Output Type of Class Library cannot be started directly due to errors in Visual Studio's project configuration`. Let me know what you think.

**Comment by alexfjw** [comment]
Maybe you missed this line in the Technical Troubleshooting Guide.
```This guide serves as a collection of technical issues/solutions commonly seen when trying to install PowerPointLabs.```This bug affects only developers, and not users so I don't think it should go in there. 

Maybe https://github.com/PowerPointLabs/PowerPointLabs/blob/master/doc/CommonTraps.md might be a better place. @junkiattan Any thoughts?

**Comment by junkiattan** [comment]
Hey @YuPeiHenry 

Thanx for the update to the documentation and for the alternative solution! What @alexfjw says make sense. When I drafted the Technical Troubleshooting Guide, I also linked to it from the PowerPointLabs website. I intended it to be used by the public so that they can self-answer their queries.

For all developmental related "gotchas", it would be better to place it in [Common Development Traps](https://github.com/PowerPointLabs/PowerPointLabs/blob/master/doc/CommonTraps.md) instead, so that future developers like you can avoid such problems when developing for PowerPointLabs!

When updating the document, it would be awesome if you could include both the default solution, and any alternative solutions that you discover so others can benefit from it as well. Thank you!

**Comment by YuPeiHenry** [comment]
@junkiattan Thanks for pointing out my mistake. I have moved it to CommonTraps.md and included the alternate solution. I have also place an image of the error in the doc (not sure if this is neccessary).

File: doc/CommonTraps.md [hunk 1]
```diff
@@ -34,3 +34,14 @@ This document details the common pitfalls when developing for PowerPointLabs, so
 1. Event triggering order of `WindowDeactivated` and `PresentationClose` is different between PowerPoint 2010 and 2013. In 2010, `PresentationClose` will be triggered before `WindowDeactivated`, but this order is reversed in 2013
 
 1. Accessing the `ActivePresentation.Saved` property on a new, unsaved presentation, always results in the `ActivePresentation.Saved` property returning `msoFalse`, regardless of whether the new presentation contains any valid changes to be saved.
+
+1. Visual Studio may show the error "Output Type of Class Library cannot be started directly", shown below. This is caused by errors in Visual Studio's project configuration.
+
+![Alt text](../doc/images/CommonTraps/output-type-class-library.png)
```
Comment (alexfjw) [inline doc/CommonTraps.md]: The image and following text don't seem to line up well. Could you add tabs like in item 12 of the list?

image

**Comment by YuPeiHenry** [comment]
Thanks Alex for the suggestion on how the readability can be improved. I have added tabs.

**Comment by alexfjw** [comment]
LGTM. 
Just noticed that you're merging to master. 
Our dev branch is dev-release. Do swap to that branch instead.

**Comment by YuPeiHenry** [comment]
I think I read somewhere that documentation updates are done on the main branch? Sure, I will change to dev-release branch once I have access to my computer.

**Comment by alexfjw** [comment]
Unfortunately our `master` branch is slightly ahead of `dev-release` and your PR contains some changes from `master`. Could you rebase onto `dev-release` in the meantime? We can merge this PR once that's done.

**Pull request merged**